Bump vLLM examples to 0.14.0 and unify CUDA/Python base images

mmangkad opened the pull request:
- bump every vllm example to latest vllm versions
- standardize all vllm-serving images on cuda 12.8.1 registry bases
- tune the gpt-oss example for blackwell b200 gpus with the recommended optimized settings
- migrate the cuda graph configuration to the new `--max-cudagraph-capture-size` api

charlesfrye commented:
Thanks for the PR @mmangkad! And apologies for the delayed reply, I was OOO and then at NeurIPS.

I wasn't able to get the GPT-OSS example working. I had to make some changes to the cmd construction to get vllm to start. After starting, the server hangs after logging the cubin loads from FlashInfer (see next message).

You can find my code [here](https://github.com/modal-labs/modal-examples/tree/charlesfrye/vllm-upgrades).

Could you please open a separate PR with just the changes to the other examples and then we can figure out the GPT-OSS issues concurrently?

mmangkad commented:
Hey @charlesfrye, thanks for the feedback and apologies too for the delay!

I've updated the PR:

- Bumped all vLLM examples to 0.13.0
- Updated base images to cu129 since it is now the default for vLLM
- GPT-OSS: Switched from B200 to H200 - the FlashInfer MXFP4 MoE kernel on Blackwell really takes too long to compile (10-15+ min), which causes the hang you observed. I'm suggesting to use Hopper again to avoid this issue of very long startup time.

Let me know your thoughts on this.
